ksqldb-rocksdb-config-setter/src/main/java/io/confluent/ksql/rocksdb/KsqlBoundedMemoryRocksDBConfig.java

```diff
@@ -92,7 +117,34 @@ public class KsqlBoundedMemoryRocksDBConfig extends AbstractConfig {
           Type.DOUBLE,
           INDEX_FILTER_BLOCK_RATIO_DEFAULT,
           Importance.LOW,
-          INDEX_FILTER_BLOCK_RATIO_DOC
+          INDEX_FILTER_BLOCK_RATIO_DOC)
+      .define(
+          COMPACTION_STYLE,
+          Type.STRING,
+          CompactionStyle.UNIVERSAL.name(),
+          ConfigDef.ValidString.in(
+              CompactionStyle.UNIVERSAL.name(), CompactionStyle.LEVEL.name()),
+          Importance.LOW,
+          COMPACTION_STYLE_DOC)
+      .define(
+          COMPRESSION_TYPE,
+          Type.STRING,
+          CompressionType.NO_COMPRESSION.name(),
+          ConfigValidators.enumValues(CompressionType.class),
+          Importance.LOW,
+          COMPRESSION_TYPE_DOC
+      ).define(
+          MAX_BACKGROUND_JOBS,
```

Comment:
What's the relation between this and the existing `NUM_BACKGROUND_THREADS` config? Is the concern that the default number of concurrent background jobs may be the current limitation, even if the number of background threads is increased?

Reply:
> What's the relation between this and the existing `NUM_BACKGROUND_THREADS` config? Is the concern that the default number of concurrent background jobs may be the current limitation, even if the number of background threads is increased?

Adding this option was based on @cadonna 's suggestion (based on a [study](http://smalldatum.blogspot.com/2022/07/the-impact-of-maxbackgroundjobs-on.html)). The [docs](https://github.com/EighteenZi/rocksdb_wiki/blob/master/Thread-Pool.md) show that `NUM_BACKGROUND_THREADS` and `MAX_BACKGROUND_JOBS` are more or less the same!
The client must create a thread pool by setting the number of background threads, but the recommended way to to configure background compaction and flush parallelism is with `MAX_BACKGROUND_JOBS`!